### PR TITLE
[converters/stack.py] Update stack converter to support dynamic shapes. Resolves #830.

### LINKED ISSUE #830: [Bug] `torch.stack` fails for dynamic shapes with error `Error Code 7: Internal Error (IShuffleLayer :0:SHUFFLE:GPU: reshaping failed for tensor: input_0 reshape would change volume`
`torch.stack` appears to not yet support dynamic shapes.

Specifically, the following script fails:
```
  import logging
  import tensorrt
  import torch
  import torch2trt


  logging.basicConfig(level=logging.INFO)
  torch.manual_seed(0)

  DEVICE = torch.device("cuda:0")


  class Model(torch.nn.Module):
      def forward(self, a):
          return torch.stack((a, a))


  if __name__ == "__main__":
      tensor = torch.rand((2,2), dtype=torch.float32, device=DEVICE)

      model = Model().eval().to(DEVICE)
      model_trt = torch2trt.torch2trt(
          model, [tensor], log_level=tensorrt.Logger.INFO, min_shapes=[(1,1)], max_shapes=[(5,5)]
      )

      out = model(tensor)
      print(f'Out {out}')

      out_trt = model_trt(tensor)
      print(f'Out TRT {out_trt}')

      assert torch.allclose(out, out_trt)

      tensor = torch.rand((5,5), dtype=torch.float32, device=DEVICE)

      out = model(tensor)
      print(f'Out {out}')

      out_trt = model_trt(tensor)
      print(f'Out TRT {out_trt}')

      assert torch.allclose(out, out_trt)
```
with the following error:
```
[01/24/2023-13:13:55] [TRT] [E] 7: [shapeMachine.cpp::execute::702] Error Code 7: Internal Error (IShuffleLayer :0:SHUFFLE:GPU: reshaping failed for tensor: input_0
reshape would change volume
Instruction: RESHAPE_ZERO_IS_PLACEHOLDER{5 5} {1 2 2}
)
```

This was run in a conda environment with the following settings:
python 3.10
pytorch 1.13.1
nvidia-tensorrt 8.4.3.1
torch2trt (latest master)

### PR DESCRIPTION
**Pull request by chaoz-dev**
Resolves issue #830, where `torch.stack` does not currently correctly handle dynamic shapes during TRT conversion.

Checked with included test:
```
python torch2trt/converters/stack_test.py
```

### CODE REVIEW

**Comment by jaybdub** [comment]
Thanks for the PR!

I believe when I was working on dynamic shapes for either stack or chunk, I had an issue where adding dynamic shape support broke static models.

It's been a while so I forget what the concern was exactly, but I may want to sanity check a few things first.
